app/src/main/java/org/jellyfin/androidtv/ui/startup/StartupActivity.kt

```diff
@@ -0,0 +1,158 @@
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import org.jellyfin.androidtv.R
+import org.jellyfin.androidtv.TvApp
+import org.jellyfin.androidtv.data.model.LoadingState
+import org.jellyfin.androidtv.data.model.ServerList
+import org.jellyfin.androidtv.data.model.User
+import org.jellyfin.androidtv.ui.browsing.MainActivity
+import org.jellyfin.androidtv.ui.itemdetail.FullDetailsActivity
+import org.jellyfin.androidtv.ui.itemhandling.ItemLauncher
+import org.jellyfin.androidtv.ui.playback.MediaManager
+import org.jellyfin.androidtv.util.Utils
+import org.jellyfin.androidtv.util.apiclient.toUserDto
+import org.jellyfin.apiclient.interaction.ApiClient
+import org.jellyfin.apiclient.interaction.Response
+import org.jellyfin.apiclient.model.dto.BaseItemDto
+import org.koin.android.viewmodel.ext.android.viewModel
+import org.koin.java.KoinJavaComponent.get
+import timber.log.Timber
+
+class StartupActivity : FragmentActivity() {
+	companion object {
+		private const val NETWORK_PERMISSION = 1
+		const val ITEM_ID = "ItemId"
+		const val ITEM_IS_USER_VIEW = "ItemIsUserView"
+		const val HIDE_SPLASH = "HideSplash"
+	}
+
+	private var application: TvApp? = null
+	private val loginViewModel: LoginViewModel by viewModel()
+	private var isLoaded = false
+
+	override fun onCreate(savedInstanceState: Bundle?) {
+		super.onCreate(savedInstanceState)
+		setContentView(R.layout.fragment_content_view)
+		if (!intent.getBooleanExtra(HIDE_SPLASH, false)) {
+			supportFragmentManager.beginTransaction()
+				.replace(R.id.content_view, SplashFragment())
+				.commit()
+		}
+		application = applicationContext as TvApp
+
+		//Ensure basic permissions
+		if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M && (ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_NETWORK_STATE) != PackageManager.PERMISSION_GRANTED
+						|| ContextCompat.checkSelfPermission(this, Manifest.permission.INTERNET) != PackageManager.PERMISSION_GRANTED)) {
+			Timber.i("Requesting network permissions")
+			ActivityCompat.requestPermissions(this, arrayOf(Manifest.permission.ACCESS_NETWORK_STATE, Manifest.permission.INTERNET), NETWORK_PERMISSION)
+		} else {
+			Timber.i("Basic network permissions are granted")
+
+			val loadingObserver = Observer<ServerList> { serverList ->
+				Timber.d("LoadingState: %s", serverList.state.toString())
+				if (serverList.state == LoadingState.SUCCESS && !isLoaded) start()
+			}
+			loginViewModel.serverList.observe(this, loadingObserver)
+		}
+
+		// Navigate to home screen when user has logged in
+		val currentUserObserver = Observer<User> { user: User? ->
+			// TODO: This should be removed in favor of fragments getting the current user directly
+			application!!.currentUser = user?.toUserDto()
+			// User has been logged in continue to home screen
+			// TODO: We should use a fragment for this, but the current fragment depends on BaseActivity
+			val intent = Intent(application, MainActivity::class.java)
```

Comment:
We probably wan't to use `openNextActivity()` here because the user might have opened the app with an intent but required authentication first.